src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteEmbeddedLanguage/FrameworkParametersCompletionProvider.cs

```diff
@@ -0,0 +1,594 @@
+    }
+
+    private static bool HasNonRouteAttribute(SyntaxToken token, SemanticModel semanticModel, WellKnownTypes wellKnownTypes, CancellationToken cancellationToken)
+    {
+        if (token.Parent?.Parent is ParameterSyntax parameter)
+        {
+            foreach (var attributeList in parameter.AttributeLists.OfType<AttributeListSyntax>())
+            {
+                foreach (var attribute in attributeList.Attributes)
+                {
+                    var attributeTypeSymbol = semanticModel.GetSymbolInfo(attribute, cancellationToken).GetAnySymbol();
+
+                    if (attributeTypeSymbol != null)
+                    {
+                        foreach (var nonRouteMetadataType in wellKnownTypes.NonRouteMetadataTypes)
+                        {
+                            if (attributeTypeSymbol.ContainingSymbol is ITypeSymbol typeSymbol &&
+                                typeSymbol.Implements(nonRouteMetadataType))
+                            {
+                                return true;
+                            }
+                        }
+                        if (SymbolEqualityComparer.Default.Equals(attributeTypeSymbol.ContainingSymbol, wellKnownTypes.AsParametersAttribute))
+                        {
+                            return true;
+                        }
+                    }
+                }
+            }
+        }
+
+        return false;
+    }
+
+    private static bool IsCurrentParameterBindable(SyntaxToken token, SemanticModel semanticModel, CancellationToken cancellationToken)
+    {
+        if (token.Parent.IsKind(SyntaxKind.PredefinedType))
+        {
+            return true;
+        }
+
+        var parameterTypeSymbol = semanticModel.GetSymbolInfo(token.Parent, cancellationToken).GetAnySymbol();
+        if (parameterTypeSymbol is INamedTypeSymbol typeSymbol)
+        {
+            if (typeSymbol.SpecialType == SpecialType.System_String)
+            {
+                return true;
+            }
+
+            // Check if the parameter type has a static TryParse method.
+            foreach (var item in typeSymbol.GetMembers("TryParse"))
+            {
+                if (item is IMethodSymbol methodSymbol &&
+                    methodSymbol.IsStatic &&
+                    methodSymbol.ReturnType.SpecialType == SpecialType.System_Boolean &&
+                    methodSymbol.Parameters.Length == 2 &&
+                    methodSymbol.Parameters[0].Type.SpecialType == SpecialType.System_String &&
+                    methodSymbol.Parameters[1].RefKind == RefKind.Out)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+        else if (parameterTypeSymbol is IMethodSymbol)
+        {
+            // If the parameter type is a method then the method is bound to the minimal API.
+            return false;
+        }
+
+        // Token could the identifier with completion explicitly triggered.
```

Comment:
What is this trying to say?

Reply:
The cursor could be on an identifier and completion is explicitly triggered (e.g. CTRL+SPACE)